client/app/bundles/course/assessment/question/programming/actions/onlineEditorActionCreators.jsx

```diff
@@ -33,25 +33,81 @@ export function deletePythonTestCase(testType, index) {
   };
 }
 
-export function updateNewDataFile(filename, index) {
+export function updatePythonNewDataFile(filename, index) {
   return {
     type: actionTypes.PYTHON_NEW_DATA_FILE_UPDATE,
     index,
     filename,
   };
 }
 
-export function deleteNewDataFile(index) {
+export function deletePythonNewDataFile(index) {
   return {
     type: actionTypes.PYTHON_NEW_DATA_FILE_DELETE,
     index,
   };
 }
 
-export function deleteExistingDataFile(filename, toDelete) {
+export function deletePythonExistingDataFile(filename, toDelete) {
   return {
     type: actionTypes.PYTHON_EXISTING_DATA_FILE_DELETE,
     filename,
     toDelete,
   };
 }
+
+export function updateCppCodeBlock(field, newValue) {
```

Comment:
As of now i can't think of whether java will require different kinds of actions, though it is likely that i will try to design the package to fit the current implementation of the editor, so probably would be better to stick with a common set of actions altogether.

In that case we can just refactor this back into a common action that all language-editors use. Initially i didn't know much about what would be needed for the C++ autograder so i left it like this, and then refactor later if there isn't much of a difference.